src/quemb/kbe/fragment.py

```diff
@@ -237,34 +234,36 @@ def fragmentate(
     elif frag_type == "chemgen":
         if kpt is None:
             raise ValueError("Provide kpt mesh in fragmentate() and restart!")
-        if n_BE != 1:
-            raise ValueError(
-                "Only be_type='be1' is currently supported for periodic chemgen!"
-            )
+        if additional_args is None:
+            additional_args = ChemGenArgs()
         else:
-            warn("Periodic BE1 with chemgen is a temporary solution.")
+            assert isinstance(additional_args, ChemGenArgs)
         fragments = chemgen(
-            mol.to_mol(),
+            mol,
             n_BE=n_BE,
             frozen_core=frozen_core,
-            args=ChemGenArgs(),
+            args=additional_args,
             iao_valence_basis=iao_valence_basis,
         )
-        molecular_FragPart = fragments.get_FragPart()
         if print_frags:
             print(fragments.frag_structure.get_string())
+        # Once periodic FragPart API is fixed,
+        # add _get_FragPart_no_iao equivalent in quemb.molbe.chemgen
+        mol_fragments = fragments.get_FragPart(
```

Comment:
It might be worth a thought to not allow the `_wrong_iao_indexing` in the periodic case. In the molecular part we only have it for testing purposes of previous code, but AFAIK there is no previous working periodic, IAO code.

Reply:
How about we revisit this once we have a clearer picture of #125? Not allowing `_wrong_iao_indexing` should be rather simple at that point

Reply:
Yes, makes absolutely sense